config/300-kafkasource.yaml

```diff
@@ -0,0 +1,385 @@
+# Copyright 2022 TriggerMesh Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: kafkasources.sources.triggermesh.io
+  labels:
+    eventing.knative.dev/source: 'true'
+    duck.knative.dev/source: 'true'
+    knative.dev/crd-install: 'true'
+    triggermesh.io/crd-install: 'true'
+  annotations:
+    registry.knative.dev/eventTypes: |
+      [
+        { "type": "io.triggermesh.kafka.message" }
+      ]
+spec:
+  group: sources.triggermesh.io
+  scope: Namespaced
+  names:
+    kind: KafkaSource
+    plural: kafkasources
+    categories:
+    - all
+    - knative
+    - eventing
+    - sources
+  versions:
+  - name: v1alpha1
+    served: true
+    storage: true
+    subresources:
+      status: {}
+    schema:
+      openAPIV3Schema:
+        description: TriggerMesh event target for Kafka.
+        type: object
+        properties:
+          spec:
+            description: Desired state of the event source.
+            type: object
+            properties:
+              topics:
+                description: Topic name to stream the target events to.
+                type: array
+                items:
+                  type: string
+                  minLength: 1
+              bootstrapServers:
+                description: Array of Kafka servers used to bootstrap the connection.
+                type: array
+                items:
+                  type: string
+                  minLength: 1
+              groupID:
+                description: The ID of the kafka group.
+                type: string
+              salsEnable:
```

Comment:
Im not going to make any holdups here! So this is not a request for this PR. but i think we could improve the organisation of this CRD a bit by including an auth section like so :
```
auth
auth.securityMechanism.
auth.sasl.username
auth.sasl.password
auth.kerberos
auth.kerberos.realm
auth.kerberos.serviceName
auth.kerberos.username
auth.kerberos.password
auth.kerberos.config
auth.kerberos.keytab
auth.ssl.sslCA
auth.ssl.clientCert
auth.ssl.clientKey
...
```

Reply:
here is a better explaination -> https://github.com/triggermesh/triggermesh/blob/0b516e9d217092f7f73eb0085f290c1a9c47579a/config/300-kafkasource.yaml#L70-L243